src/Makefile

```diff
@@ -50,6 +68,22 @@ $V/poh_verify.o: $(POH_SRCS)
 	@mkdir -p $(@D)
 	$(NVCC) -rdc=true $(CFLAGS) -c $< -o $@
 
+$V/cl_crypt-dlink.o: $V/cl_init_platform.o $V/cl_verify.o
+	ar rvs $@ $^
+
+$V/lib$(CL_LIB).a: $V/cl_crypt-dlink.o $V/cl_init_platform.o $V/cl_verify.o
+	ar rcs $@ $^
+
+$V/lib$(CL_LIB).so: $V/cl_crypt-dlink.o $V/cl_init_platform.o $V/cl_verify.o
+	g++ -shared --shared $^ -o $@
+
+$V/cl_ecc_main.o: $(CL_ECC_DIR)/main.cpp $(ECC_DIR)/ed25519.h
+	@mkdir -p $(@D)
+	g++ $(CL_CFLAGS) -I$(ECC_DIR) -c $< -o $@
+
+$V/$(CL_ECC_TEST_BIN): $V/cl_ecc_main.o $V/lib$(CL_LIB).a
+	g++ $(CL_CFLAGS) -Wl,-v -L/usr/local/cuda-9.1/lib64 -L$V -lpthread $^ -lOpenCL -o $@
```

Comment:
Need to do something cleaner than a hardcoded path to cuda here. Maybe `CUDA_HOME?=/usr/local/cuda` then `-L$(CUDA_HOME)/lib64`